### ISSUE TITLE
Migrate to Libadwaita/GTK4

### ISSUE THREAD

**Issue by imhemish** [mijorus/webapp-manager#1]
Migrating the app to Gtk4/Libadwaita can help it do adapt to interfaces of mobile devices, and also work well in tiling window managers. The GUI of codebase of this app is small, so it would be easy enought to port the interface. I may work on it after 1-2 months, but currently I am busy, so just opening the issue to get reminded of this task

**Comment by imhemish** [comment]
![Workbench screenshot](https://user-images.githubusercontent.com/74466492/218302338-c581cbb2-25db-41af-8302-57f93199e51b.png)
A sneak peek of how the interface might look

**Comment by imhemish** [comment]
I am also working on editing the WebAppManagerWindow.py to adapt to all these changes and use GTK4

**Comment by mijorus** [comment]
Hi @imhemish,
I'm currently not planning to migrate this app to adwaita (personally), but I would be happy to accept your PRs.

 Furthermore, like the README says, there are some "issues" I would like to solve before re-submitting this app to Flathub.

**Comment by imhemish** [comment]
> Hi @imhemish, I'm currently not planning to migrate this app to adwaita (personally), but I would be happy to accept your PRs.
> 
> Furthermore, like the README says, there are some "issues" I would like to solve before re-submitting this app to Flathub.

Yeah you work on that, I am not touching common.py at all. All backend problems would be fixable in common.py
I am not expecting you to migrate it to gtk4. I would happily do that myself and make a PR

**Comment by mijorus** [comment]
Detecting when a user uninstalls a web browser, especially, is challenging.
One solution could be re-write how apps are launched, basically Webapp Manager should always be in the middle somehow.

Or I might creare a custom run command that does something before actually launching the app, will see...

**Comment by imhemish** [comment]
> Detecting when a user uninstalls a web browser, especially, is challenging. One solution could be re-write how apps are launched, basically Webapp Manager should always be in the middle somehow.
> 
> Or I might creare a custom run command that does something before actually launching the app, will see...

We can maybe run the app in daemon mode to detect changes, which ofcourse the user could disabled to run in daemon mode

**Comment by mijorus** [comment]
Yes, but which change should I detect?

This app supports flatpaks, snaps, system packages...
